Core/Modules/CANMessageLookUpModule.c

```diff
@@ -40,3 +42,14 @@ void MotorTemperatureCallback(iCommsMessage_t msg)
 {
 	DebugPrint("MotorTemperatureCallback! %d", msg.standardMessageID);
 }
+
+PUBLIC const ICommsMessageInfo* CANMessageLookUpGetInfo(ICommsMessageLookUpIndex id) {
```

Comment:
You don't need to iterate through the array, using the id as the index should return the correct row of information for that can message.